sklearn/preprocessing/target.py

```diff
@@ -0,0 +1,222 @@
+# Authors: Guillaume Lemaitre <guillaume.lemaitre@inria.fr>
+#          Andreas Mueller <amueller@ais.uni-bonn.de>
+# License: BSD 3 clause
+
+import numpy as np
+
+from ..base import BaseEstimator, RegressorMixin, clone, is_regressor
+from ..linear_model import LinearRegression
+from ..utils.validation import check_is_fitted
+from ._function_transformer import FunctionTransformer
+
+__all__ = ['TransformedTargetRegressor']
+
+
+class TransformedTargetRegressor(BaseEstimator, RegressorMixin):
+    """Meta-estimator to apply a transformation to the target before fitting.
+
+    Useful for applying a non-linear transformation in regression
+    problems. This transformation can be given as a Transformer such as the
+    QuantileTransformer or as a function and its inverse such as ``np.log`` and
+    ``np.exp``.
+
+    The computation during ``fit`` is::
+
+        estimator.fit(X, func(y))
+
+    or::
+
+        estimator.fit(X, transformer.transform(y))
+
+    The computation during ``predict`` is::
+
+        inverse_func(estimator.predict(X))
+
+    or::
+
+        transformer.inverse_transform(estimator.predict(X))
+
+    Parameters
+    ----------
+    estimator : object, (default=LinearRegression())
+        Estimator object derived from ``RegressorMixin``.
```

Comment:
we don't usually require inheritance, as long as appropriate methods are available. You could say "such as derived from" ...

Perhaps mention that it will be cloned.